### ISSUE TITLE
Updating Ingress annotations are not picked up by controller

### ISSUE THREAD

**Issue by julianvmodesto** [appscode/voyager#115]
Currently, when I add or update annotations to my Ingress object, none of the changes are made.

For example, before:

```
apiVersion: appscode.com/v1beta1
kind: Ingress
metadata:
  name: frontend
  labels:
    app: frontend
spec:
  # Default backend
  backend:
    serviceName: echoserver
    servicePort: 80
```

After:
```
apiVersion: appscode.com/v1beta1
kind: Ingress
metadata:
  name: frontend
  labels:
    app: frontend
  annotations:
    # Enable the AWS Proxy Protocol
    # NOTE(jmodes): We need to have `bind :80 accept-proxy` for HAProxy to accept Proxy Protocol requests.
    # ingress.appscode.com/annotations.service: '{"service.beta.kubernetes.io/aws-load-balancer-proxy-protocol": "*"}'
    # Number of HAProxy replicas to run
    ingress.appscode.com/replicas: 1
    # Open HAProxy stats on port 1936
    ingress.appscode.com/stats: true
    ingress.appscode.com/stats.secretName: frontend-stats
spec:
  # Default backend
  backend:
    serviceName: echoserver
    servicePort: 80
```

After applying the new Ingress, none of the changes were enacted by the controller.

**Comment by TamalSaha** [comment]
Fixed ini #134

**Comment by julianvmodesto** [comment]
Unfortunately, I think i'm running into this issue again on v5.0.0-rc.11.

My change is adding a new annotation.

```
# Old
    ingress.appscode.com/annotations-service: |
      {
        "service.beta.kubernetes.io/aws-load-balancer-ssl-cert": "REDACTED",
        "service.beta.kubernetes.io/aws-load-balancer-ssl-ports": "443",
        "service.beta.kubernetes.io/aws-load-balancer-backend-protocol": "http"
      }

# New
    ingress.appscode.com/annotations-service: |
      {
        "service.beta.kubernetes.io/aws-load-balancer-ssl-cert": "REDACTED",
        "service.beta.kubernetes.io/aws-load-balancer-ssl-ports": "443",
        "service.beta.kubernetes.io/aws-load-balancer-backend-protocol": "http",
        "external-dns.alpha.kubernetes.io/hostname": "REDACTED"
      }
```

**Comment by tamalsaha** [comment]
If you retry does it work?

https://github.com/appscode/voyager/pull/803 will permanently solve this type of issues, hopefully.

**Comment by julianvmodesto** [comment]
Ohh hmm sorry yes after retrying it did pick up, weird!

Interesting, never implemented a controller before, hope that the workqueue is helpful.

**Comment by tamalsaha** [comment]
We have merged #803.
